### ISSUE TITLE
Mass transfer case labelling wrong (most notably, Case C mass transfer being labelled as Case B)

### ISSUE THREAD

**Issue by TomWagg** [TeamCOMPAS/COMPAS#483]
**Describe the bug**
_tldr: I think HG::DetermineMassTransferCase is overwriting GiantBranch::DetermineMassTransferCase such that case C mass transfer is always labelled as case B. But there seem to be more issues with the labelling other than this..._

@LiekeVanSon presented some work to us today and pointed out that COMPAS seems to be mislabelling mass transfer cases. Here are some lovely plots made by her that illustrate the point.
<img width="50%" src="https://user-images.githubusercontent.com/21990332/102116671-4b128780-3e0b-11eb-86e8-e5966e9f916e.png">
In this plot each binary is coloured based on the mass transfer case of each star in the binary (i.e. A B, means primary is case Case, secondary is case B). As you can see, there are no pink/purple points. The unique cases that show up in the results are `[‘- -’ ‘- A’ ‘- B’ ‘A B’ ‘B -’ ‘B B’]`. To further illustrate this, here is another of Lieke's plots that shows the detailed evolution of a binary labelled ["-, B"] (none, case B)
<img width="50%" src="https://user-images.githubusercontent.com/21990332/102116971-b65c5980-3e0b-11eb-91a3-d4279f6dc4e1.png">
It seems from this plot that the mass transfer for each star seems to be occurring after CHeB starts and so should actually be case CC.

We looked into this further together and it seems that due to the way the inheritance works in COMPAS, the HG::DetermineMassTransferCase function is overwriting the GiantBranch::DetermineMassTransferCase function and leading to all case C being labelled case B. I think this can be fixed by moving GiantBranch::DetermineMassTransferCase to CHeB::DetermineMassTransferCase!

This does **not** explain why in the second plot, case C mass transfer is being labelled as None. So there may be further issues.

Lieke and I were also puzzled as to why HeMS and HeHG are referred to as Case A and Case B mass transfer?

**Label the issue**
`urgency_low`,`severity_moderate`

**To Reproduce**
Run a grid of binaries and check the unique mass transfer cases, case C never shows up

**Expected behavior**
Stars of CHeB and beyond should be case C.

**Versioning (please complete the following information):**
 - COMPAS - v02.17.11
